src/components/EditableHeading/EditableHeading.tsx

```diff
@@ -1,244 +1,166 @@
-    return <Heading {...contentProps} />;
-  };
-
-  const getInputProps = () => {
-    const textAreaType = props.ellipsisMaxLines > 1 ? InputType.TEXT_AREA : undefined;
-    const inputType = props.inputType || textAreaType;
-    return {
-      value: valueState,
-      className: cx(
-        getStyle(headingStyles, camelCase("element-type-" + type)),
-        getStyle(headingStyles, camelCase("size-" + size)),
-        inputClassName
-      ),
-      isValidValue: props.isValidValue,
-      onChange: props.onChange,
-      onKeyDown: props.onKeyDown,
-      onKeyPress: props.onKeyPress,
-      onClick: props.onClick,
-      customColor,
-      onTabHandler: props.onTabHandler,
-      onArrowKeyDown: props.onArrowKeyDown,
-      autoComplete: props.autoComplete,
-      maxLength: props.maxLength,
-      placeholder: props.placeholder,
-      shouldFocusOnMount: props.shouldFocusOnMount,
-      selectOnMount: props.selectOnMount,
-      onBlur: props.onBlur,
-      onFocus: props.onFocus,
-      inputType,
-      ignoreBlurClass: props.ignoreBlurClass,
-      autoSize,
-      textareaSubmitOnEnter: props.textareaSubmitOnEnter,
-      onFinishEditing: onFinishEditingCallback,
-      onCancelEditing: onCancelEditingCallback,
-      onIgnoreBlurEvent: onIgnoreBlurEventCallback,
-      onError: onInputErrorCallback,
-      onSuccess: onInputSuccessCallback,
-      ariaLabel: props.inputAriaLabel
-    };
-  };
-
-  const renderInputComponent = () => {
-    const inputProps = getInputProps();
-    // eslint-disable-next-line react/jsx-props-no-spreading
-    return <EditableInput {...inputProps} />;
-  };
-
-  const shouldEdit = !disabled && isEditing;
-
-  return (
-    <div
-      ref={ref}
-      style={style}
-      className={cx(styles.editableHeadingWrapper, className, {
-        [styles.insetFocus]: insetFocus
-      })}
-      aria-label={`${value} ${tooltip || ""}`}
-      id={id}
-      data-testid={overrideDataTestId || getTestId(ComponentDefaultTestId.EDITABLE_HEADING, id)}
-    >
-      <Clickable role={shouldEdit ? "button" : "input"} onClick={onClick} disabled={disabled}>
-        {shouldEdit ? renderInputComponent() : renderContentComponent()}
-      </Clickable>
-    </div>
-  );
-};
+
+    useLayoutEffect(() => {
+      if (focusOnMount) {
+        focus();
+      }
+      if (selectOnMount) {
+        select();
+      }
+    }, [focusOnMount, isEditing, selectOnMount]);
+
+    const inputClassNames = useMemo(() => {
+      return cx(styles.input, getStyle(styles, camelCase(type + "-" + weight)));
+    }, [type, weight]);
+
+    return (
+      <div
+        ref={mergedRef}
+        id={id}
+        aria-label={ariaLabel}
+        data-testid={dataTestId || getTestId(ComponentDefaultTestId.EDITABLE_HEADING, id)}
+        className={cx(styles.editableHeading, className)}
+      >
+        <HiddenInputPlaceholder className={inputClassNames} value={inputValue} onChange={setInputWidth} />
```

Comment:
The main difference between this and the old one is that the input is now an inline element and not a black element, so since html inputs cannot have a dynamic width that fits the text we use this hack of a hidden placeholder that defines the width for the input. This is a very common hack and it's being used in libraries such as `react-input-autosize`.